File: backend/app/schemas/trailer.py
from datetime import date, datetime
from pydantic import BaseModel, ConfigDict


class TrailerBase(BaseModel):
    numero_economico: str
    placas: str
    marca: str | None = None
    modelo: str | None = None
    anio: int | None = None
    poliza_seguro: str | None = None
    seguro_vigencia: date | None = None
    tarjeta_circulacion: str | None = None
    tarjeta_vigencia: date | None = None
    verificacion: str | None = None
    verificacion_vigencia: date | None = None
    activo: bool = True


class TrailerCreate(TrailerBase):
    pass


class TrailerUpdate(BaseModel):
    numero_economico: str | None = None
    placas: str | None = None
    marca: str | None = None
    modelo: str | None = None
    anio: int | None = None
    poliza_seguro: str | None = None
    seguro_vigencia: date | None = None
    tarjeta_circulacion: str | None = None
    tarjeta_vigencia: date | None = None
    verificacion: str | None = None
    verificacion_vigencia: date | None = None
    activo: bool | None = None


class TrailerResponse(TrailerBase):
    id_trailer: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/trailer.ts
export type Trailer = {
  id_trailer: number;
  numero_economico: string;
  placas: string;
  marca: string | null;
  modelo: string | null;
  anio: number | null;
  poliza_seguro: string | null;
  seguro_vigencia: string | null;
  tarjeta_circulacion: string | null;
  tarjeta_vigencia: string | null;
  permiso_circulacion: string | null;
  numero_serie: string | null;
  verificacion: string | null;
  verificacion_vigencia: string | null;
  activo: boolean;
  created_at: string;
  updated_at: string;
};

export type CreateTrailerPayload = {
  numero_economico: string;
  placas: string;
  marca?: string | null;
  modelo?: string | null;
  anio?: number | null;
  poliza_seguro?: string | null;
  seguro_vigencia?: string | null;
  tarjeta_circulacion?: string | null;
  tarjeta_vigencia?: string | null;
  permiso_circulacion?: string | null;
  numero_serie?: string | null;
  verificacion?: string | null;
  verificacion_vigencia?: string | null;
  activo: boolean;
};

export type UpdateTrailerPayload = {
  numero_economico?: string | null;
  placas?: string | null;
  marca?: string | null;
  modelo?: string | null;
  anio?: number | null;
  poliza_seguro?: string | null;
  seguro_vigencia?: string | null;
  tarjeta_circulacion?: string | null;
  tarjeta_vigencia?: string | null;
  permiso_circulacion?: string | null;
  numero_serie?: string | null;
  verificacion?: string | null;
  verificacion_vigencia?: string | null;
  activo?: boolean;
};
